### PR TITLE
P1703R1 Recognizing header unit imports requires full preprocessing

### LINKED ISSUE #3003: [2019-07 CWG Motion 24] P1703R1 Recognizing Header Unit Imports Requires Full Preprocessing

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3003.

### CODE REVIEW

File: source/preprocessor.tex [hunk 1]
```diff
@@ -662,104 +666,39 @@
...
-and the \tcode{import} token is not within
-the \grammarterm{header-name-tokens} or \grammarterm{pp-import-suffix}
-of another \grammarterm{pp-import}.
-The \tcode{;} \grammarterm{preprocessing-token}
-terminating a \grammarterm{pp-import}
-shall not have been produced by
-macro replacement\iref{cpp.replace}.
-The \defnadj{point of}{macro import} for a \grammarterm{pp-import} is
-immediately after the \tcode{;} terminating the \grammarterm{pp-import}.
+The \defnadj{point of}{macro import} for the
```
Comment (burblebee) [inline source/preprocessor.tex]: I know this was in the original text, but I think this definition should be changed to: `\defnadj{point of import}{macro}`.
Likewise, while it's outside the scope of this paper, the \defnx{}s under "macro definition" should also be moved under "macro", i.e.: 
`\defnx{point of definition}{macro definition!point of definition}`
should become:
`\defnadj{point of definition}{macro}`

File: source/preprocessor.tex [hunk 2]
```diff
@@ -844,6 +783,49 @@
 
 \indextext{macro import|)}%
 
+\rSec1[cpp.glob.frag]{Global module fragment}
+
+\begin{bnf}
+\nontermdef{pp-global-module-fragment}\br
+    \terminal{module} \terminal{;} pp-balanced-token-seq \terminal{module}
+\end{bnf}
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/preprocessor.tex]: Fixed.

File: source/lex.tex [hunk 3]
```diff
@@ -338,17 +339,13 @@
 except that a \grammarterm{header-name}\iref{lex.header} is only formed
 \begin{itemize}
 \item
-within a \tcode{\#include} directive\iref{cpp.include},
+after the \tcode{include} or \tcode{import} preprocessing token in a
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/lex.tex]: Fixed.

File: source/preprocessor.tex (hunk 1 above)
Comment (jensmaurer) [reply source/preprocessor.tex]: We can only use \defnadj{x}{y} for stuff that should read "x y" in the resulting document, but I get your point.

I've changed stuff.

Comment (jensmaurer) [reply source/preprocessor.tex]: Everything "macro" is now under "macro".
Everything "point of "is now under "point of" with cross-references.

Comment (burblebee) [reply source/preprocessor.tex]: > We can only use \defnadj{x}{y} for stuff that should read "x y" ...
Duh, yeah :).  Glad you got my point.  Thanks for the fixes.

File: source/basic.tex [hunk 4]
```diff
@@ -916,6 +919,8 @@
 see~\ref{temp.point}.\end{note}%
 \indextext{scope!declarations and|)}
```
Comment (burblebee) [inline source/basic.tex]: remove newline?

File: source/preprocessor.tex [hunk 5]
```diff
@@ -739,7 +742,8 @@
 \end{itemize}
 
 \pnum
-A macro directive is \defnx{active}{macro definition!active} at a source location
+\indextext{active macro directive|see{macro, active}}%
```
Comment (burblebee) [inline source/preprocessor.tex]: Not sure this index is needed.

File: source/basic.tex (hunk 4 above)
Comment (jensmaurer) [reply source/basic.tex]: Sure.

File: source/preprocessor.tex (hunk 5 above)
Comment (jensmaurer) [reply source/preprocessor.tex]: Well, we do define "active macro directive" in this sentence, so it seems plausible to have an index entry for that exact phrase, although we'd like people to look up the meat under "macro".

**Pull request merged**